src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -2871,7 +2877,19 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 
 	//#region Editor Contributions
 	getContribution<T extends INotebookEditorContribution>(id: string): T {
-		return <T>(this._contributions.get(id) || null);
+		const entry = this._contributions.get(id);
+		if (!entry) {
+			return null!; // TODO
+		}
+
+		if (entry instanceof SyncDescriptor) {
```

Comment:
This guarantees "slowness" when calling `getContribution` and I would suggest to use `IdleValue` because that is more optimistic and usually lucky. Workbench contributions that are "eventual" use the same fundamentals (runWhenIdle)